PhysicsTools/NanoAOD/plugins/BToKstllProducer.cc

```diff
@@ -1407,7 +1445,7 @@ void BToKstllProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup
 	for (unsigned int k = 0; k < (pfCandNumber+lostChHadrTrackNumber); ++k) {
 	  if(debug) std::cout << " i = " << i << " leptonNumber = " << leptonNumber << " j = " << j << " k = " << k << std::endl; 
 
-	  if(!isLep2PFL && (j-leptonNumber) == k) continue;
+	  if( (isLep1LowPt == false && isLep1PFL == false && (i-leptonNumber) == k)  || (isLep2LowPt == false && isLep2PFL == false && (j-leptonNumber) == k) ) continue;
```

Comment:
change with 
if( ( (isLep1PFC || isLep1LT) && (i-leptonNumber) == k) ||
    ( (isLep2PFC || isLep2LT) && (j-leptonNumber) == k) ) continue;

(check that this is correct. Also your version is correct but less logic)